File: backend/internal/models/expense.go
package models

type CreateExpenseInput struct {
	Amount     *int   `json:"amount"`
	CategoryID *int   `json:"category_id"`
	Memo       string `json:"memo"`
	SpentAt    string `json:"spent_at"`
	Status     string `json:"status"`
}

type UpdateExpenseInput struct {
	ID         int    `json:"id"`
	Amount     *int   `json:"amount"`
	CategoryID *int   `json:"category_id"`
	Memo       string `json:"memo"`
	SpentAt    string `json:"spent_at"`
	Status     string `json:"status"`
}

type Expense struct {
	ID       int      `json:"id"`
	Amount   int      `json:"amount"`
	Memo     string   `json:"memo"`
	SpentAt  string   `json:"spent_at"`
	Status   string   `json:"status"`
	Category Category `json:"category"`
}

File: apps/web/src/lib/types/expense.ts
export type CreateExpenseInput = {
    amount: number
    category_id: number
    memo?: string
    spent_at: string // yyyy-mm-dd
    status?: 'planned' | 'confirmed'
}

export type UpdateExpenseInput = {
    amount: number
    category_id: number
    memo?: string
    spent_at: string // yyyy-mm-dd or date-time
    status?: 'planned' | 'confirmed'
}

export type Expense = {
    id: number;
    amount: number;
    memo: string | null;
    spent_at: string; // YYYY-MM-DD
    status: 'planned' | 'confirmed';
    category: {
        id: number;
        name: string;
    }
};

export type GetExpensesResponse = {
    expenses: Expense[];
};